Makefile: "@if [ -t 1 ]; then tput setaf ..." breaks make lint

```
% gmake lint TESTS=TestGolint
GOPATH set to /data/home/kena/src/go
Running make with -j16
Running make with -j16
gmake[1]: Nothing to be done for 'build_lib_static'.
[100%] Built target libprotobuf
[100%] Built target snappy
[  0%] Built target build_version
[100%] Built target rocksdb
[100%] Built target roach
[ 86%] Built target roach
[100%] Built target roachccl
GEN pkg/cli/zcgo_flags.go
GEN pkg/server/status/zcgo_flags.go
GEN pkg/storage/engine/zcgo_flags.go
GEN pkg/ccl/storageccl/engineccl/zcgo_flags.go
NOTE: `make lint` is very slow! Perhaps `make lintshort`?
gmake: *** [Makefile:341: lint] Error 1
```

The problem is (I believe) that the last command in the if ... fi block fails (`tput sgr0` is not valid on this system). The following patch fixes:

```
- @if [ -t 1 ]; then tput setaf 3 2>/dev/null; echo 'NOTE: `make lint` is very slow! Perhaps `make lintshort`?'; tput sgr0 2>/dev/null; fi
+ @if [ -t 1 ]; then tput setaf 3 2>/dev/null; echo 'NOTE: `make lint` is very slow! Perhaps `make lintshort`?'; tput sgr0 2>/dev/null; fi || true
```

--- a/Makefile
+++ b/Makefile
@@ -338,7 +338,7 @@ generate: ## Regenerate generated code.
 .PHONY: lint
 lint: override TAGS += lint
 lint: ## Run all style checkers and linters.
-	@if [ -t 1 ]; then tput setaf 3 2>/dev/null; echo 'NOTE: `make lint` is very slow! Perhaps `make lintshort`?'; tput sgr0 2>/dev/null; fi
+	@if [ -t 1 ]; then echo $$(tput setaf 3 2>/dev/null)'NOTE: `make lint` is very slow! Perhaps `make lintshort`?'$$(tput sgr0 2>/dev/null); fi
 	$(XGO) test ./build -v $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LINKFLAGS)' -run 'TestStyle/$(TESTS)'
 
 .PHONY: lintshort

